### ISSUE TITLE
Cannot find way to mock nested function calls.

### ISSUE THREAD

**Issue by timbrown5** [jenkinsci/JenkinsPipelineUnit#353]
### Version report
`jenkinsVersion = '2.222.1'`
```
    compile group: 'org.codehaus.groovy', name: 'groovy-all', version: '2.5.10'
    compile group: 'junit', name: 'junit', version: '4.12'

    // Jenkins base test harness
    testCompile 'org.jenkins-ci.main:jenkins-test-harness:2.62'

    // Jenkins Pipeline JUnit
    // Upgrading 'jenkins-pipeline-unit' to 1.4 makes the test take a lot longer
    // (~1 minute to ~ 8 minutes). It could be an incompatibility with
    // 'PipelineSpockTestBase' as that was added to allow some testing of
    // Declarative Pipelines - and 1.4 have some support for that too.
    compile group: 'com.lesfurets', name:'jenkins-pipeline-unit', version:'1.3'
    compile group: 'com.cloudbees', name: 'groovy-cps', version: '1.12'
    compile group: 'org.jenkins-ci.main', name: 'jenkins-core', version: '2.85'
    compile group: 'org.jenkins-ci.plugins', name: 'plugin', version: '1.499'
```

### Reproduction steps
`vars/sayHello.groovy`:
```
def call(String name) {
    echo "Hello, ${name}"
}
```

`vars/doMore.groovy`:
```
def call(String name) {
    echo "Doing somrthing fist"
    if(name) {
        sayHello(name)
    }
}
```

`DoMoreTests.groovy`:
```
package test.unit.vars

import ai.graphcore.BasePipelineSpecification

/*groovylint-disable ClassJavadoc, JUnitPublicNonTestMethod, MethodName */
class DoMoreTests extends BasePipelineSpecification {

    private doMore = null

    def setup() {
        super.setUp()
        doMore = loadScript('vars/doMore.groovy')
        helper.registerAllowedMethod('sayHello.echo', [String], null)
    }

    def "test doMore no name"() {
        when:
        doMore()

        then:
        assertJobStatusSuccess()
    }

    def "test doMore with name"() {
        when:
        doMore("Joe Bloggs")

        then:
        assertJobStatusSuccess()
    }
}
```

### Results

Expected result:

Both tests pass.

Actual result:
`test doMore with name` fails with:
```
groovy.lang.MissingMethodException: No signature of method: sayHello.echo() is applicable for argument types: (org.codehaus.groovy.runtime.GStringImpl) values: [Hello, Joe Bloggs]
Possible solutions: each(groovy.lang.Closure), macro(groovy.lang.Closure), wait(), run(), run(), every()
	at sayHello.call(sayHello.groovy:2)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.lesfurets.jenkins.unit.PipelineTestHelper$_setGlobalVars_closure10.closure20$_closure21(PipelineTestHelper.groovy:416)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.lesfurets.jenkins.unit.PipelineTestHelper.callClosure(PipelineTestHelper.groovy:509)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.lesfurets.jenkins.unit.PipelineTestHelper$_closure3.doCall(PipelineTestHelper.groovy:149)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at doMore.call(doMore.groovy:4)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.lesfurets.jenkins.unit.PipelineTestHelper.callMethod(PipelineTestHelper.groovy:167)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.lesfurets.jenkins.unit.PipelineTestHelper$_closure3.doCall(PipelineTestHelper.groovy:154)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at test.unit.vars.DoMoreTests.test doMore with name(DoMoreTests.groovy:26)
```

**Comment by timbrown5** [comment]
Seems to be fixed in 1.4 (thought I checked this before, but obviously not 😬).

That said:
Version 1.3:
```
$ time gradle test
... 
real    0m48.401s
user    0m1.857s
sys     0m0.215s
```
vs version 1.4 (second run, so dependencies should be there):
```
$ time gradle test
... 
real    5m14.095s
user    0m2.205s
sys     0m0.380s
```
vs version 1.9 (second run, so dependencies should be there):
```
$ time gradle test
... 
real    5m8.380s
user    0m2.265s
sys     0m0.399s
```

**Comment by timbrown5** [comment]
Just to follow up, I fixed my test slowness as described in this issue: https://github.com/macg33zr/pipelineUnit/issues/18
